pkg/k8s/apis/cilium.io/v2/validator/unknown_fields.go

```diff
@@ -0,0 +1,236 @@
+			return err
+		}
+
+		var err error
+		filtered, err = toUnstructured(ccnp, true)
+		if err != nil {
+			return err
+		}
+	default:
+		// We've already validated above that there can only be two kinds: CNP &
+		// CCNP.
+	}
+
+	given, err := getFields(policy.Object)
+	if err != nil {
+		return err
+	}
+
+	expected, err := getFields(filtered.Object)
+	if err != nil {
+		return err
+	}
+
+	// Compare the expected policy with the given policy to find all unknown
+	// fields.
+	var r reporter
+	if !cmp.Equal(
+		expected,
+		given,
+		cmp.Reporter(&r),
+		cmpopts.SortSlices(func(o, n string) bool {
+			return o < n
+		}),
+	) {
+		scopedLog.Warn(warnUnknownFields)
+		return ErrUnknownFields{
+			extras: r.extras,
+		}
+	}
+
+	return nil
+}
+
+const (
+	warnTopLevelDescriptionField = "It seems you have a policy with a " +
+		"top-level description. This field is no longer supported. Please migrate " +
+		"your policy's description field under `spec` or `specs`."
+
+	warnUnknownFields = "It seems you have a policy with extra unknown fields. " +
+		"Consider removing these fields, as they have no effect. The presence " +
+		"of these fields may have introduced a false sense security, so please " +
+		"check whether your policy is actually behaving as you expect."
+)
+
+// ErrTopLevelDescriptionFound is the error returned if a policy contains a
+// top-level description field. Instead this field should be moved to under
+// (Rule).Description.
+var ErrTopLevelDescriptionFound = errors.New("top-level description field found")
+
+// ErrUnknownFields is an error representing the condition where unknown fields
+// were found within a policy during validation. Fields that are not expected
+// to be in the policy will be put inside the "extras" slice.
+type ErrUnknownFields struct {
+	extras []string
+}
+
+func (e ErrUnknownFields) Error() string {
+	return fmt.Sprintf("unknown fields found, extra:%v", e.extras)
+}
+
+func toUnstructured(obj interface{}, clusterwide bool) (*unstructured.Unstructured, error) {
+	var (
+		b   []byte
+		err error
+	)
+	if clusterwide {
+		b, err = json.Marshal(obj.(*cilium_v2.CiliumClusterwideNetworkPolicy))
+	} else {
+		b, err = json.Marshal(obj.(*cilium_v2.CiliumNetworkPolicy))
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	r := new(unstructured.Unstructured)
+	if err := r.UnmarshalJSON(b); err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+func getFields(u map[string]interface{}) ([]string, error) {
+	flat, err := flattenObject(u)
+	if err != nil {
+		return nil, err
+	}
+
+	fields := make([]string, 0, len(flat))
+	for k := range flat {
+		if !igIgnoredField(k) {
+			fields = append(fields, k)
+		}
+	}
+
+	return fields, nil
+}
+
+func flattenObject(obj map[string]interface{}) (map[string]interface{}, error) {
+	return flatten.Flatten(obj, "", flatten.DotStyle)
+}
+
+func igIgnoredField(k string) bool {
+	return k == "metadata.creationTimestamp" || strings.HasPrefix(k, "spec.endpointSelector.matchLabels")
+}
```

Comment:
Maybe add explanation why these are ignored?

Reply:
and should it be `isIgnoredField` instead of `igIgnoredField` ?

Reply:
:man_facepalming: How did I not notice that. :sweat_smile: